hmt_escrow/storage.py

```diff
@@ -58,25 +47,53 @@ def download(key: str, private_key: bytes) -> Dict:
     Raises:
         Exception: if reading from IPFS fails.
 
+    >>> credentials = {
+    ... 	"gas_payer": "0x1413862C2B7054CDbfdc181B83962CB0FC11fD92",
+    ... 	"gas_payer_priv": "28e516f1e2f99e96a48a23cea1f94ee5f073403a1c68e818263f0eb898f1c8e5"
+    ... }
+    >>> pub_key = b"2dbc2c2c86052702e7c219339514b2e8bd4687ba1236c478ad41b43330b08488c12c8c1797aa181f3a4596a1bd8a0c18344ea44d6655f61fa73e56e743f79e0d"
+    >>> job = Job(credentials=credentials, escrow_manifest=manifest)
+    >>> (hash_, manifest_url) = upload(job.serialized_manifest, pub_key)
+    >>> manifest_dict = download(manifest_url, job.gas_payer_priv)
+    >>> manifest_dict == job.serialized_manifest
+    True
     """
+
+    # Q prefix is ipfs. else assume ipns
+    if key[0] != 'Q':
+        res = IPFS_CLIENT.resolve(f'/ipns/{key}')
+        ipfs_path = res['Path']
+        key = ipfs_path.split('/')[-1]
     try:
         LOG.debug("Downloading key: {}".format(key))
         ciphertext = IPFS_CLIENT.cat(key)
     except Exception as e:
         LOG.warning(
-            "Reading the key {} with private key {} with IPFS failed because of: {}"
-            .format(key, private_key, e))
+            f'Reading the key {str(key)} with private key {str(private_key)} with IPFS failed because of: {str(e)}'
+        )
         raise e
     msg = _decrypt(private_key, ciphertext)
     return json.loads(msg)
 
 
 @timeout_decorator.timeout(20)
-def upload(msg: Dict, public_key: bytes) -> Tuple[str, str]:
+def upload(msg: Dict, public_key: bytes,
+           ipns_keypair_name: str = '') -> Tuple[str, str]:
     """Upload and encrypt a string for later retrieval.
     This can be manifest files, results, or anything that's been already
     encrypted.
 
+    Args:
```

Comment:
Might also want to add another note here that describes what it does with IPNS: ie "additionally creates a IPNS link to the uploaded file hash that gets updated as uploads change" (if thats accurate)